src/utils/getProject.ts

```
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfProject } from '@salesforce/core';
+
+/*
+ * Get the sfdx project from the current dir.
+ * It will return `undefined` if there's no project.
+ * */
+export async function maybeGetProject(): Promise<SfProject | undefined> {
+  try {
+    return await SfProject.resolve();
+  } catch (err) {
+    if (err instanceof Error && err.name === 'InvalidProjectWorkspaceError') {
+      return undefined;
+    }
```

any swallow all other errors? 
What's the logic of checking the error name then?

---

> What's the logic of checking the error name then?

none, I think I was just following the error msg from `SfProject.resolve`'s jsdoc. 
Changed to make it swallow all errors that could come from trying to resolve a project.